org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/MessageUtils.java

```diff
@@ -0,0 +1,114 @@
+package org.eclipse.wildwebdeveloper.debug;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.internal.ui.DebugUIPlugin;
+import org.eclipse.debug.internal.ui.launchConfigurations.LaunchConfigurationEditDialog;
+import org.eclipse.debug.internal.ui.launchConfigurations.LaunchGroupExtension;
+import org.eclipse.debug.ui.DebugUITools;
+import org.eclipse.debug.ui.ILaunchConfigurationTab;
+import org.eclipse.debug.ui.ILaunchGroup;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.preference.IPreferenceNode;
+import org.eclipse.jface.preference.PreferenceDialog;
+import org.eclipse.jface.preference.PreferenceManager;
+import org.eclipse.jface.preference.PreferenceNode;
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.BusyIndicator;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.internal.browser.WebBrowserPreferencePage;
+import org.eclipse.wildwebdeveloper.Activator;
+
+@SuppressWarnings("restriction")
+public class MessageUtils {
+	private static final String BROWSER_TAB_NAME = "Browser"; //$NON-NLS-1$
+	private static final String WEB_BROWSER_PREFERENCE_PAGE = "org.eclipse.ui.browser.preferencePage"; //$NON-NLS-1$
+	
+	private MessageUtils() {
+		// stateless, don't instantiate
+	}
+	
+	public static int showBrowserLocationsConfigurationError(ILaunchConfiguration configuration, String mode, String browserName, boolean suggestEditLaunch) {
+		String title = MessageFormat.format(Messages.RuntimeExecutable_NotDefinedError_Title, browserName);
+		String browserTabMessage = suggestEditLaunch ? MessageFormat.format(Messages.RuntimeExecutable_NotDefinedError_Message_1, browserName) : ""; //$NON-NLS-1$
+		String webBrowsersMessage = MessageFormat.format(Messages.RuntimeExecutable_NotDefinedError_Message_2, browserName);
+		String message = MessageFormat.format(Messages.RuntimeExecutable_NotDefinedError_Message, browserName, browserTabMessage, webBrowsersMessage);
+		IStatus errorStatus = new Status(IStatus.ERROR, Activator.PLUGIN_ID, title);
+		Activator.getDefault().getLog().log(errorStatus);
+
+		List<String> buttons = new ArrayList<>();
+		if (suggestEditLaunch) {
+			buttons.add(Messages.RuntimeExecutable_NotDefinedError_OpenLaunchConfigurationBrowserTab);
+		}
+		buttons.add(Messages.RuntimeExecutable_NotDefinedError_OpenWebBrowserPreferencePage);
+		buttons.add(IDialogConstants.CANCEL_LABEL);
+		
+		int result[] = {Window.CANCEL};
+		Display.getDefault().syncExec(() -> {
+			int response = suggestEditLaunch ?
+					MessageDialog.open(MessageDialog.ERROR, Activator.getShell(), title, message, SWT.NONE, 
```

Comment:
The shell should be a method parameter as with current implementation, using multiple screens, the dialog can show on the other monitor.

Reply:
OK